### PR TITLE
Greg/cherry/deposit

### PR DESCRIPTION
**Pull request by GregTheGreek**
Addresses bugs in #200

### CODE REVIEW

**Review by wemeetagain — DISMISSED** [review]

File: src/eth1/impl/ethers.ts [hunk 1]
```diff
@@ -89,13 +90,22 @@ export class EthersEth1Notifier extends EventEmitter implements Eth1Notifier {
     this.emit('block', block);
   }
 
-  public async processDepositLog(dataHex: string, indexHex: string): Promise<void> {
-    const deposit = this.createDeposit(dataHex, indexHex);
+  public async processDepositLog(pubkey: string, withdrawalCredentials: string, amount: string, signature: string, merkleTreeIndex: string): Promise<void> {
+    const deposit: Deposit = {
+      proof: Array.from({length: DEPOSIT_CONTRACT_TREE_DEPTH},() => Buffer.alloc(32)),
+      index: (await this.contract.from_little_endian_64(merkleTreeIndex)).toNumber(),
```
Comment (wemeetagain) [inline src/eth1/impl/ethers.ts]: I think you should use BN or ssz
This is breaking CI

Comment (mpetrunic) [reply src/eth1/impl/ethers.ts]: contract is undefined in test, it should be stubbed. we could write our own little to big endian conversion.

Comment (GregTheGreek) [reply src/eth1/impl/ethers.ts]: Interesting, I'm getting different errors running locally. Debugging...

**Comment by mpetrunic** [comment]
@GregTheGreek I fixed the tests. Bloody hell it was a mess with those changes. Generally, we should keep deposit contract defaults in one place and in dev defaults, just extend with different bytecode

File: src/db/api/beacon/beacon.ts [hunk 2]
```diff
@@ -217,7 +217,11 @@ export class BeaconDB extends DatabaseService implements IBeaconDb {
   }
 
   public async deleteGenesisDeposits(deposits: Deposit[]): Promise<void> {
-    await this.deleteData(Bucket.genesisDeposit, Deposit, deposits);
+    const criteria: any[] = [];
+    deposits.forEach((deposit) =>
```
Comment (wemeetagain) [inline src/db/api/beacon/beacon.ts]: can be shortened to
`const criteria = deposits.map(...`

File: src/eth1/impl/ethers.ts [hunk 3]
```diff
@@ -113,22 +122,27 @@ export class EthersEth1Notifier extends EventEmitter implements Eth1Notifier {
...
-      depositCount,
-    };
-    const genesisDeposits = await this.genesisDeposits();
-    this.emit('eth2genesis', time, genesisDeposits, genesisEth1Data);
-    //from now on it will be kept in BeaconBlock
-    await this.db.deleteGenesisDeposits(genesisDeposits);
+    try {
+      const depositRoot = Buffer.from(depositRootHex.substr(2), 'hex');
+      const depositCount = Buffer.from(depositCountHex.substr(2), 'hex').readUIntLE(0, 6);
+      const time = new BN(Buffer.from(timeHex.substr(2), 'hex').readUIntLE(0, 6));
```
Comment (wemeetagain) [inline src/eth1/impl/ethers.ts]: I think time should be a `number`.

File: src/eth1/wallet.ts [hunk 4]
```diff
@@ -2,35 +2,75 @@
...
-  public async createValidatorDeposit(address: string, amount: BigNumber): Promise<string> {
-    let contract = new ethers.Contract(address, defaults.depositContract.abi, this.wallet);
-    //TODO: Implement real deposit arguments according to spec : https://github.com/ethereum/eth2.0-specs/blob/dev/specs/core/0_beacon-chain.md#deposit-arguments
-    const depositData = Buffer.alloc(512);
-    const tx: ContractTransaction = await contract.deposit(depositData, {value: amount});
-    await tx.wait();
-    return tx.hash;
+  public async createValidatorDeposit(address: string, value: BigNumber): Promise<string> {
+    // TODO Regardless of the return value, the calling function will assume success, need to error handle.
+    // Minor hack, no real performance loss
```
Comment (wemeetagain) [inline src/eth1/wallet.ts]: remove comment

**Review by wemeetagain — CHANGES_REQUESTED** [review]

File: src/eth1/impl/ethers.ts (hunk 3 above)
Comment (GregTheGreek) [reply src/eth1/impl/ethers.ts]: Yep. should be of type number 64

**Review by wemeetagain — APPROVED** [review]

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**